docs/resource/os.md

```diff
@@ -14,7 +14,9 @@ In case of virtualized environments, this is the operating system as it is obser
 | `os.type` | string | The operating system type. | `windows` | Required |
 | `os.description` | string | Human readable (not intended to be parsed) OS version information, like e.g. reported by `ver` or `lsb_release -a` commands. | `Microsoft Windows [Version 10.0.18363.778]`; `Ubuntu 18.04.1 LTS` | Recommended |
 | `os.name` | string | Human readable operating system name. | `iOS`; `Android`; `Ubuntu` | Recommended |
-| `os.version` | string | The version string of the operating system as defined in [Version Attributes](/docs/resource/README.md#version-attributes). | `14.2.1`; `18.04.1` | Recommended |
+| `os.version` | string | The major version string of the operating system.  Additionally, build numbers can be added in os.build attribute. | `14.2.1`; `18.04.1` | Recommended |
+| `os.build` | string | Unique identifier for a particular build or compilation of software. | `TQ3C.230805.001.B2`; `20E247` | Recommended |
+| `os.sdk_version` | string | Identifies the set of development tools and APIs used for building the software. | `26`; `6.1.1` | Recommended |
```

Comment:
I'm not opposed to introducing `os.sdk_version`, but wondering if that is a generic concept or something Android-specific, or mobile-specific? If so, we should at least add a note that it's not relevant for other OS families. Or, if it's mobile-specific, maybe it would make sense to move it to a new namespace `mobile.*` together with other mobile-specific resource attributes?

> **os.sdk_version**
> Additionally, there is no attribute to identify the SDK version (set of development tools & APIs used for development). It's not always 1 to 1 with the version number and is required to troubleshoot issues. Example - As seen [here](https://source.android.com/docs/setup/about/build-numbers#platform-code-names-versions-api-levels-and-ndk-releases) in android version number 12 can be associated with both API level 31 & 32.

Reply:
I think it does apply to more than mobile platforms. For example: Android, iOS, MacOS, Windows would all have a SDK version which could be different from the OS version. But I agree to the point that it might not apply to all platforms. Does it suffices to add another line to the description that says "It might not apply to all platforms."?

Reply:
I am not convinced that `sdk_version` should be under the `os` namespace, and SDK seems like something that not only applies to OSs but also to other kinds of software (e.g. OTel has an SDK concept).

Even if we were to add it under `os`, I would name it `os.sdk.version` for consistency with the existing `telemetry.sdk.version`

Reply:
@mx-psi Thanks for your feedback! Can you please elaborate why you think it should not go under os namespace? According to my understanding as it's an os sdk version it belongs to os namespace. I like the idea of keeping it consistent with telemetry.sdk.version and naming it os.sdk.version.

Reply:
> Can you please elaborate why you think it should not go under os namespace? Any suggestions on where it can go instead?

With the current description *"set of development tools and APIs used for building the software"* this could also apply to any non-OS specific SDK (e.g. the OpenTelemetry SDKs, the AWS SDKs, the Flutter SDK). If the attribute is meant to be this general and apply to any of the previous examples I don't think it should be under the `os` namespace. Even restricting us to OS-specific SDKs, it also seems like the are multiple SDKs for the same OS version, at least for Windows where we have the [Windows SDK](https://en.wikipedia.org/wiki/Microsoft_Windows_SDK) and the [Windows App SDK](https://en.wikipedia.org/wiki/Windows_App_SDK), so it seems like the version is not enough to specify the SDK even when knowing the OS.

Reply:
@mx-psi Thanks for clarifying! 

I think as the concept of an OS sdk version does apply to more than mobile platforms, having it here might help all platforms. But we should definitely define it in a way that's not confusing. Can the approach suggested below work: 

- Naming it os.sdk.version
- Changing description to include the words "operating system" instead of "software" ->  "set of development tools and APIs used for building the operating system" (All other descriptions in the file include those words, I missed it)

For special cases like the one you mentioned where Windows can have multiple SDK versions:
-  Windows can define this string's value to include both. Example: **"Windows SDK 1.2.3, Windows App SDK 3.4.5"** 
- We can add that in description of this attribute to clarify - "If OS has multiple relevant SDKs, all of them can be specified as follows - **"<Name> <Version>, <Name> <Version>"**. Example - "Windows SDK 1.2.3, Windows App SDK 3.4.5" " 

(or ";" as a separator instead of ",")

Reply:
> Changing description to include the words "operating system" instead of "software" -> "set of development tools and APIs used for building the operating system" (All other descriptions in the file include those words, I missed it)

Makes sense to me to clarify this

> For special cases like the one you mentioned where Windows can have multiple SDK versions:

An alternative would be to clarify which one we are talking about with the description. Do you think there is a way to do this? I am unfortunately not familiar with the Windows SDK

Reply:
We can give multiple examples (the ones we're clear on - iOS, MacOS, Android) and then mention a generic statement that says "If there are multiple relevant SDKs for a given OS, all of them can be specified as follows - "[Name] [Version], [Name] [Version]". " 

Complete description could be: 
Identifies the set of development tools and APIs used for building the operating system. If OS has only one specific SDK, directly the version number is specified, like for Android API level 26, just the value "26" suffices. If OS has multiple relevant SDKs, all of them can be specified as follows - "[Name] [Version], [Name] [Version]". 

And then the Examples could contain [`26`; `6.1.1` ; `Windows SDK 1.2.3, Windows App SDK 3.4.5`]

I'll try to verify the windows example with someone from that platform. But does the idea look good to you? I also plan on bringing this up in next Monday's semantic conventions WG meeting at 8am PST.

Reply:
Using the words "operating system" does indeed make it clear that Windows and macOS would have very limited scope here. Only Microsoft and Apply employees will know what it means to talk about the "SDK" to build the Windows OS or the Apple macOS, since presumably only those employees would even have access to build the operating system.

Even for Android the scope would be limited to a very small subset of engineering groups who actually build the Android operating system, albeit in this case the groups would not be limited to just Google employees, since everyone can get the sources to build an Android operating system.

I do agree with @mx-psi that the term SDK is not specific to operating systems.

Reply:
Thanks all for the review! As per our discussion in semantic-conventions WG meeting and after discussion in client sig meeting, we've decided to add an android specific attribute to capture sdk version - "android.os.api.level". Will create a PR for that next. Keeping this PR for just os.build. 

@AlexanderWert @mx-psi @cwegener Please do approve this PR!